bpmn-statistic/src/com/axonivy/utils/bpmnstatistic/managedbean/ProcessesAnalyticsBean.java

```diff
@@ -0,0 +1,214 @@
+  private TimeIntervalFilter timeIntervalFilter;
+  private ProcessMiningData processMiningData;
+  private String selectedPid;
+  private String miningUrl;
+  private ContentObject processMiningDataJsonFile;
+  private String bpmnIframeSourceUrl;
+
+  @PostConstruct
+  private void init() {
+    processesMap = ProcessUtils.getProcessesWithPmv();
+    applicationName = Ivy.request().getApplication().getName();
+    targetHost = Ivy.html().applicationHomeLink().toAbsoluteUri().getAuthority();
+    targetApplicationName = applicationName;
+    setNodes(new ArrayList<>());
+    processMiningDataJsonFile = ContentManagement.cms(IApplication.current()).root().child()
+        .folder(ProcessAnalyticsConstants.PROCESS_MINING_CMS_PATH).child()
+        .file(ProcessAnalyticsConstants.DATA_CMS_PATH, ProcessAnalyticsConstants.JSON_EXTENSION);
+    miningUrl = processMiningDataJsonFile.uri();
+  }
+
+  public KpiType[] getKpiTypes() {
+    return KpiType.values();
+  }
+
+  public List<String> getAvailableProcesses() {
+    if (StringUtils.isBlank(selectedModule)) {
+      return new ArrayList<>();
+    }
+    return processesMap.get(selectedModule).stream().map(IWebStartable::getDisplayName).collect(Collectors.toList());
+  }
+
+  public Set<String> getAvailableModules() {
+    return processesMap.keySet();
+  }
+
+  public void onModuleSelect() {
+    if (StringUtils.isBlank(selectedModule)) {
+      selectedProcess = null;
+      setTotalFrequency(0);
+    }
+    resetStatisticValue();
+  }
+
+  public void onProcessSelect() {
+    resetStatisticValue();
+  }
+
+  public void onKpiTypeSelect() {
+    resetStatisticValue();
+  }
+
+  private void resetStatisticValue() {
+    processMiningData = null;
+    nodes = new ArrayList<>();
+    bpmnIframeSourceUrl = "";
+  }
+
+  public void onShowStatisticBtnClick() {
+    loadNodes();
+    updateProcessMiningDataJson();
+    updateBpmnIframeSourceUrl();
+  }
+
+  private IProcessWebStartable getSelectedIProcessWebStartable() {
+    return CollectionUtils.emptyIfNull(processesMap.get(selectedModule)).stream()
+        .filter(process -> process.getDisplayName().equalsIgnoreCase(selectedProcess)).findAny().orElse(null);
+  }
+
+  private void updateBpmnIframeSourceUrl() {
+    String processFilePath = getSelectedIProcessWebStartable().getId()
+        .replace(targetApplicationName + '/' + selectedModule + '/', "");
+    int lastSlashIndex = processFilePath.lastIndexOf('/');
+
+    if (lastSlashIndex != -1) {
+      processFilePath = processFilePath.substring(0, lastSlashIndex) + ".p.json";
+    }
+    bpmnIframeSourceUrl = String.format(ProcessAnalyticsConstants.BPMN_STATISTIC_SOURCE_URL_PATTERN, applicationName,
+        targetHost, targetApplicationName, selectedModule, processFilePath);
+  }
+
+  private void loadNodes() {
+    totalFrequency = 0;
+    if (StringUtils.isNotBlank(selectedProcess) && StringUtils.isNotBlank(selectedModule) && selectedKpiType != null) {
+      if (timeIntervalFilter == null) {
```

Comment:
init time interval when the UI is loaded to avoid this checking every "loadNodes"